api/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the networking.olm v1alpha1 API group
+//+kubebuilder:object:generate=true
+//+groupName=networking.olm.openshift.io
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "networking.olm.openshift.io", Version: "v1alpha1"}
```

Comment:
Doesn't `networking` look too general for this operator? Like it it was part of a bigger networking OLM API.

Reply:
What do you recommend?

Reply:
The `IngressController` belongs to the group `config.openshift.io`. My thinking here was the types are part of the `networking` group which is distributed through `olm` on `openshift.io`.

Reply:
> What do you recommend?

Hm, good question.
I don't see a lot of examples in `olm.openshift.io` domain, only our ExternalDNS which is `externaldns.olm.openshift.io`. MetalLB Operator is in its own group: `metlalb.io`.

Something like `aws-lb-controller.networking.olm.openshift.io` indeed looks not as redundant as `externaldnses.externaldns.olm.openshift.io`.

I'm just wondering how such a group can be coordinated if tomorrow we add Azure Gateway Ingress Controller, for instance. Will we have to have to upgrade the APIs of all the controllers to a new version (e.g. `v1alpha1` -> `v1beta1`) if just one of them changes?

Reply:
> Something like aws-lb-controller.networking.olm.openshift.io indeed looks not as redundant as externaldnses.externaldns.olm.openshift.io

That is exactly what it is now. Because the full name will be `awsloadbalancercontroller.networking.olm.openshift.io`. Just the group name is `networking.olm.openshift.io`.

Reply:
> I'm just wondering how such a group can be coordinated if tomorrow we add Azure Gateway Ingress Controller, for instance. Will we have to have to upgrade the APIs of all the controllers to a new version (e.g. v1alpha1 -> v1beta1) if just one of them changes?

No each one will be upgraded separately.